dbt/config.py

```diff
@@ -5,9 +5,6 @@
 
 
 def read_config(profiles_dir=None):
-    if profiles_dir is None:
-        profiles_dir = default_profiles_dir
-
     path = os.path.join(profiles_dir, 'profiles.yml')
```

Comment:
I think passing `None` to `os.path.join` causes a `TypeError`.

Previously, this file was assumed to be located at `~/.dbt/profiles.yml`. We should either remove the default value for `profiles_dir` and insist that the caller passes a value, or we should coalesce `None` to the default filepath